### PR TITLE
build: semantic release and new script names for builds

### PR DESCRIPTION
**Pull request by priyajeet**
- Moves build folder to conf per new folder standard
- Switches build to use semantic release
- Updates release script eliminating publish/examples script
- Uses new release branch for releases

Usage

`yarn release`
Will run a release but will reset release branch to master before doing so.

`yarn release:hotfix`
Will run a release without resetting release branch to master. In this case it is assumed some patch has been cherry picked on the release branch for a hotfix release.

`source ./conf/cut_release.sh` followed by some function in that script.
If you want to run only certain aspects of the release script.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @priyajeet has signed the CLA. Thanks for the pull request!

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #646

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: conf/prepush.sh [hunk 1]
```diff
@@ -21,9 +21,9 @@ check_and_commit_updated_translations() {
 # lint, test, and build assets to update translations
 prepush() {
     echo "--------------------------------------------------------"
-    echo "Pre-build"
+    echo "Build bundles"
     echo "--------------------------------------------------------"
-    yarn pre-build || exit 1
+    yarn build:i18n || exit 1
```
Comment (jstoffan) [inline conf/prepush.sh]: Do you need to clean just the i18n directories out first?

File: DEVELOPING.md [hunk 2]
```diff
@@ -1,55 +1,73 @@
...
 
-* .editorconfig - http://editorconfig.org/
-* .eslintignore - http://eslint.org/docs/user-guide/configuring#ignoring-files-and-directories
-* .eslintrc.js - http://eslint.org/docs/user-guide/configuring
-* .gitignore - https://git-scm.com/docs/gitignore
-* .stylelintrc - https://stylelint.io/user-guide/configuration/
-* .travis.yml - https://docs.travis-ci.com/user/customizing-the-build
-* babel.config.js - https://babeljs.io/docs/en/config-files/
-* postcss.config.js - https://github.com/postcss/postcss-loader
+To push a new release use `yarn release`.
```
[the remark this answers is not in the document]
Comment (priyajeet) [reply DEVELOPING.md]: Not necessary, since they are overwritten only if needed by checking contents

File: conf/prepush.sh (hunk 1 above)
Comment (priyajeet) [reply conf/prepush.sh]: Not necessary, since they are overwritten only if needed by checking contents

**Pull request merged**